liboverscroll/src/main/java/me/everything/android/ui/overscroll/adapters/ViewPagerOverScrollDecorAdapter.java

```diff
@@ -0,0 +1,55 @@
+package me.everything.android.ui.overscroll.adapters;
+
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.View;
+
+import me.everything.android.ui.overscroll.HorizontalOverScrollBounceEffectDecorator;
+import me.everything.android.ui.overscroll.VerticalOverScrollBounceEffectDecorator;
+
+/**
+ * Created by Bruce too
+ * On 2016/6/16
+ * At 14:51
+ * An adapter to enable over-scrolling over object of {@link ViewPager}
+ *
+ * @see HorizontalOverScrollBounceEffectDecorator
+ * @see VerticalOverScrollBounceEffectDecorator
+ */
+public class ViewPagerOverScrollDecorAdapter implements IOverScrollDecoratorAdapter {
+
+    protected final ViewPager mViewPager;
+
+    public ViewPagerOverScrollDecorAdapter(ViewPager viewPager) {
+        this.mViewPager = viewPager;
+    }
+
+    @Override
+    public View getView() {
+        return mViewPager;
+    }
+
+    @Override
+    public boolean isInAbsoluteStart() {
+        PagerAdapter adapter = mViewPager.getAdapter();
+        if (null != adapter) {
+            if (mViewPager.getCurrentItem() == 0) {
```

Comment:
I'm afraid this condition isn't good enough:
While having the first item selected holds true when the view-pager is in absolute-left, it also hold true in case mild dragging right-to-left has been initiated and yet the pager hasn't yet settled on page #1. This means that even though the pager isn't at absolute-left, the over-scroll will be enabled -- having the drag events propagating to both the over-scroll handler AND the view-pager view.
An undesired effect introduced by this error can be seen in this use-case:
1. Set pager at absolute left (page 0), fingers off the screen.
2. Drag a few pixels right-to-left.
3. Drag back left-to-right till page 0 should be in initial position.
The error: page 0 has been sliced off by page 1 - settled 'over' it.

I HAVE created a working version of the adapter of my own. If you wish, put it in your PR and i'll merge it in.

```java
package me.everything.android.ui.overscroll.adapters;

import android.support.v4.view.ViewPager;
import android.view.View;

import me.everything.android.ui.overscroll.HorizontalOverScrollBounceEffectDecorator;
import me.everything.android.ui.overscroll.VerticalOverScrollBounceEffectDecorator;

/**
 * Created by Bruce too
 * On 2016/6/16
 * At 14:51
 * An adapter to enable over-scrolling over object of {@link ViewPager}
 *
 * @see HorizontalOverScrollBounceEffectDecorator
 * @see VerticalOverScrollBounceEffectDecorator
 */
public class ViewPagerOverScrollDecorAdapter implements IOverScrollDecoratorAdapter, ViewPager.OnPageChangeListener {

    protected final ViewPager mViewPager;

    protected int mLastPagerPosition = 0;
    protected float mLastPagerScrollOffset;

    public ViewPagerOverScrollDecorAdapter(ViewPager viewPager) {
        mViewPager = viewPager;
        mViewPager.addOnPageChangeListener(this);

        mLastPagerPosition = mViewPager.getCurrentItem();
        mLastPagerScrollOffset = 0f;
    }

    @Override
    public View getView() {
        return mViewPager;
    }

    @Override
    public boolean isInAbsoluteStart() {
        return mLastPagerPosition == 0 &&
               mLastPagerScrollOffset == 0f &&
               !mViewPager.canScrollHorizontally(-1);
    }

    @Override
    public boolean isInAbsoluteEnd() {
        return mLastPagerPosition == mViewPager.getAdapter().getCount()-1 &&
               mLastPagerScrollOffset == 0f &&
               !mViewPager.canScrollHorizontally(1);
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
        mLastPagerPosition = position;
        mLastPagerScrollOffset = positionOffset;
    }

    @Override public void onPageSelected(int position) {}
    @Override public void onPageScrollStateChanged(int state) {}
}
```
You see, view pager scrolling is a bit less trivial than initially seems, cause the selected page is often out-of-sync with what's actually visible. Thus, you have to query it's exact state to figure out the exact moment for over-scrolling. Also, you have to use a listener since it has no 'getters', I'm afraid.